R/fda-table_07.R

```diff
@@ -1,81 +1,75 @@
-library(dplyr)
-library(tern)
+#' FDA Table 7: Deaths, Safety Population, Pooled Analyses
+#'
+#' @details
+#' * `adae` must contain `SAFFL`, `USUBJID`, `TRTEMFL`, `DTHFL`, `DTHCAUS`, and the variable specified by `arm_var`.
+#' * If specified, `alt_counts_df` must contain `SAFFL`, `USUBJID`, and the variable specified by `arm_var`.
+#' * Columns are split by arm. Overall population column is excluded by default (see `lbl_overall` argument).
+#' * Numbers in table represent the absolute numbers of patients and fraction of `N` for category summary rows and
+#'   fraction of `n` (number of patients in current category) for all other rows.
+#' * All-zero rows are removed by default (see `prune_0` argument).
+#'
+#' @inheritParams argument_convention
+#'
+#' @examples
+#' adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
+#' adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
+#'
+#' set.seed(1)
+#' adae$TRTEMFL <- ifelse(adae$USUBJID %in% sample(adsl$USUBJID, size = as.integer(nrow(adsl) / 3)), "N", "Y")
+#'
+#' tbl <- make_table_07(adae = adae, alt_counts_df = adsl)
+#' tbl
+#'
+#' @export
+make_table_07 <- function(adae,
+                          alt_counts_df = NULL,
+                          show_colcounts = TRUE,
+                          arm_var = "ARM",
+                          lbl_overall = NULL,
+                          prune_0 = TRUE,
+                          annotations = NULL) {
+  checkmate::assert_subset(c(
+    "SAFFL", "USUBJID", "TRTEMFL", "DTHFL", "DTHCAUS", arm_var
+  ), names(adae))
 
-# Load and pre-process data
+  adae <- adae %>%
+    filter(SAFFL == "Y", DTHFL == "Y") %>%
+    mutate(
```

Comment:
This step treats everything not flagged with "Y" as "N" (i.e. not treatment-emergent). In my opinion this was fine for the first template version. But since end-users will now only make a function call instead of having a real chance to look at the code, it might make sense to either discuss this behaviour or at least make it transparent in the roxygen documentation

Reply:
Same for line 39. Everything that is not flagged as treatment-emergent ("Y") will be treated as nontreatment-emergent. Do we want that?

This brings me to a more general question: Is there already a plan on how get feedback from potential end-users about decisions we make here? Will the PO's take care of this?